extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -344,14 +345,13 @@ private String resolveDriver(String dataSourceName, String dbKind,
             }
         }
 
-        throw new ConfigurationException("Unable to find a JDBC driver corresponding to the database kind '"
-                + dbKind + "' for the "
-                + (DataSourceUtil.isDefault(dataSourceName) ? "default datasource"
-                        : "datasource '" + dataSourceName + "'")
-                + ". Either provide a suitable JDBC driver extension, define the driver manually, or disable the JDBC datasource by adding "
-                + (DataSourceUtil.isDefault(dataSourceName) ? "'quarkus.datasource.jdbc=false'"
-                        : "'quarkus.datasource." + dataSourceName + ".jdbc=false'")
-                + " to your configuration if you don't need it.");
+        throw new ConfigurationException(String.format(
+                "Unable to find a JDBC driver corresponding to the database kind '%s' for the %s (available: '%s'). "
+                        + "Check if it's a typo, otherwise provide a suitable JDBC driver extension, define the driver manually, or disable the JDBC datasource by adding %s to your configuration if you don't need it.",
+                dbKind, DataSourceUtil.isDefault(dataSourceName) ? "default datasource" : "datasource '" + dataSourceName + "'",
+                jdbcDriverBuildItems.stream().map(JdbcDriverBuildItem::getDbKind).collect(Collectors.joining("','")),
+                DataSourceUtil.isDefault(dataSourceName) ? "'quarkus.datasource.jdbc=false'"
+                        : "'quarkus.datasource." + dataSourceName + ".jdbc=false'"));
```

Comment:
Better use quotes, I think? I assume they exist for a reason, though I don't know it myself.

```suggestion
                        : "'quarkus.datasource.\"" + dataSourceName + "\".jdbc=false'"));
```

If you feel like it, it might be a good idea to also introduce a util similar to this in `DataSourceUtils`:

https://github.com/quarkusio/quarkus/blob/56ab22e70cee3a6abd95b345490aa65728f37d1f/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRuntimeConfig.java#L42-L47

... because from what I see there are other places where this kind of logic is needed.

Reply:
I pushed the improvement in a separate commit.